XBVC/emitters/templates/Bitvec.hs

```diff
@@ -100,16 +177,16 @@ encodeFloat f = BS.concat [signEnc, wholeEnc, fracEnc]
 -- | Decode an EBV (stored in a bytestring) into a floating point
 -- number and any remaining bytes
 decodeFloat :: BS.ByteString -> Maybe (Float, BS.ByteString)
-decodeFloat bs = do
-  (signRaw, bs) <- decode bs :: Maybe (Word8, BS.ByteString)
-  (wholeRaw, bs) <- decode bs :: Maybe (Word32, BS.ByteString)
-  (fracRaw, bs) <- decode bs :: Maybe (Word32, BS.ByteString)
+decodeFloat bs = runStateT (do
```

Comment:
Weird q: Why favor the fully encapsulating parens rather than runStateT $...?

Reply:
will change